File: tests/unwrap_test.cpp
#include <iostream>
#include <vector>

#include "unwrap.h"
#include "vtkWriter.h"

typedef std::array<int, 3> triangle;

int main(int argc, char **argv) {
    // cerate vertices
    paraFEM::Vector3 p0(-1, -1, 0);
    paraFEM::Vector3 p1(1, -1, 0);
    paraFEM::Vector3 p2(1, 1, 0);
    paraFEM::Vector3 p3(-1, 1, 0);
    paraFEM::Vector3 p4(0, 0, 1);

    std::vector<paraFEM::Vector3> vertices = {p0, p1, p2, p3, p4};

    // create triangles
    std::vector<triangle> triangles;
    triangles.push_back(triangle{ {0, 1, 4} });
    triangles.push_back(triangle{ {1, 2, 4} });
    triangles.push_back(triangle{ {2, 3, 4} });
    triangles.push_back(triangle{ {3, 0, 4} });

    paraFEM::VtkWriter writer = paraFEM::VtkWriter("/tmp/paraFEM/unwrap");

    paraFEM::LscmRelax flattener(vertices, triangles, std::vector<int>());
    for (auto point: flattener.get_flat_vertices())
        std::cout << point << std::endl;
    flattener.lscm();
    for (auto point: flattener.get_flat_vertices())
        std::cout << point << std::endl;

}

File: tests/python/unwrap_test.py
import paraFEM
import os
vertices = []
triangles = []

with open(os.path.dirname(__file__) + "/half_sphere.obj") as fo:
    for i, line in enumerate(fo):
        if i < 2:
            continue
        if line[0] == "v":
            vertices.append([float(p) for p in line[2:].split()])
        if line[0] == "f":
            triangles.append([int(p) - 1 for p in line[2:].split()])

unwrapper = paraFEM.LscmRelax(vertices, triangles, [])
unwrapper.lscm()
unwrapper.relax(0.01)
print(unwrapper.sol)